src/auth/googleauth.ts

```diff
@@ -126,23 +123,36 @@ export class GoogleAuth {
   }
 
   /**
-   * Obtains the default project ID for the application.
-   * @param callback Optional callback
-   * @returns Promise that resolves with project Id (if used without callback)
+   * THIS METHOD HAS BEEN DEPRECATED.
+   * It will be removed in 3.0.  Please use getProjectId instead.
    */
   getDefaultProjectId(): Promise<string>;
   getDefaultProjectId(callback: ProjectIdCallback): void;
   getDefaultProjectId(callback?: ProjectIdCallback): Promise<string|null>|void {
+    process.emitWarning(messages.DEFAULT_PROJECT_ID_DEPRECATED);
```

Comment:
I would like this to be done like this:

```ts
const deprecation = {
  message: messages.DEFAULT_PROJECT_ID_DEPRECATED,
  code: 'google-auth-library:DEP001'
};
if (semver.satisfies(process.version, ">=8")) {
  process.emitWarning(deprecation.message, 'DeprecationWarning', deprecation.code);
} else {
  // This path can be removed once we drop support for Node 6.
  process.emitWarning(`${deprecation.code}: ${deprecation.message}`);
}
```

This addresses the comment that I added below when used with the `--throw-deprecations` flag in the client library tests.

Reply:
Awesome feedback - thank you!  I updated the code to (mostly) do this (the contract on `emitWarning` is a little different).